app/controllers/concerns/foreman_puppet/extensions/api_v2_hosts_bulk_actions_controller.rb

```diff
@@ -0,0 +1,20 @@
+module ForemanPuppet
+  module Extensions
+    module ApiV2HostsBulkActionsController
+      extend ActiveSupport::Concern
+
+      included do
+        before_action :find_editable_hosts, :only => [:change_puppet_proxy]
+
+        api :PUT, "/hosts/bulk/change_puppet_proxy", N_("Change Puppet Proxy")
+        param_group :bulk_host_ids
+        param :proxy_id, :number, :required => true, :desc => N_("ID of the Puppet proxy to reassign the hosts to")
```

Comment:
This being required means it can be used to reassign the host to a different proxy, but this cannot be used to unassign it, correct?